package/cloudshell/iac/terraform/services/provider_handler.py

```
@@ -48,49 +51,14 @@ def _set_cloud_env_vars(
-                cloud_attr_name_prefix, clp_resource_attributes, shell_helper)
+        elif clp_res_model in [AZURE1G_MODEL, AZURE2G_MODEL]:
+            clp_handler = AzureCloudProviderEnvVarHandler(clp_res_model, clp_resource_attributes, shell_helper)
 
-    @staticmethod
-    def _set_azure_env_vars_based_on_clp(azure_attr_name_prefix, clp_resource_attributes, shell_helper):
-        for attr in clp_resource_attributes:
-            if attr.Name == azure_attr_name_prefix + "Azure Subscription ID":
-                os.environ["ARM_SUBSCRIPTION_ID"] = attr.Value
-            if attr.Name == azure_attr_name_prefix + "Azure Tenant ID":
-                os.environ["ARM_TENANT_ID"] = attr.Value
-            if attr.Name == azure_attr_name_prefix + "Azure Application ID":
-                os.environ["ARM_CLIENT_ID"] = attr.Value
-            if attr.Name == azure_attr_name_prefix + "Azure Application Key":
-                dec_client_secret = shell_helper.api.DecryptPassword(attr.Value).Value
-                os.environ["ARM_CLIENT_SECRET"] = dec_client_secret
-
-    @staticmethod
-    def _set_aws_env_vars_based_on_clp(aws_attr_name_prefix, clp_resource_attributes, shell_helper):
-        dec_access_key = ""
-        dec_secret_key = ""
-        region_flag = False
-
-        for attr in clp_resource_attributes:
-            if attr.Name == aws_attr_name_prefix + "AWS Access Key ID":
-                dec_access_key = shell_helper.api.DecryptPassword(attr.Value).Value
-            if attr.Name == aws_attr_name_prefix + "AWS Secret Access Key":
-                dec_secret_key = shell_helper.api.DecryptPassword(attr.Value).Value
-            if attr.Name == aws_attr_name_prefix + "Region":
-                os.environ["AWS_DEFAULT_REGION"] = attr.Value
-                region_flag = True
-        if not region_flag:
-            raise ValueError("Region was not found on AWS Cloud Provider")
+        if clp_handler:
+            clp_handler.set_env_vars_based_on_clp()
```

in case clp_handler object is not created maybe we should raise an exception here?